src/main/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,289 @@
+        int size = matrix.length;
+        int tempJ = 1;
+        for (int i = 1; i < size - 1; i++) {
+            resultGrid[i][tempJ] = matrix[tempJ][i];
+            resultGrid[tempJ][i] = matrix[i][tempJ];
+        }
+
+        int tempI = size - 2;
+        for (int j = 1; j < size - 2; j++) {
+            resultGrid[tempI][j] = matrix[j][tempI];
+            resultGrid[j][tempI] = matrix[tempI][j];
+        }
+        return resultGrid;
+    }
+
+
+    /**
+     * mainDiagonalFlip.
+     *
+     * @param ring ring number.
+     * @return a Squarelotron.
+     */
+    Squarelotron mainDiagonalFlip(int ring) {
+        if (this.size % 2 == 1 && ring == this.numberOfRings) {
+            return this; // returns the same squarelotron if the ring number is equal to the number of rings.
+        }
+        int[][] resultGrid = mainDiagonalFlipHelper(ring);
+        return new Squarelotron(resultGrid);
+    }
+
+    /**
+     * mainDiagonalFlipHelper.
+     *
+     * @param ring ring number.
+     * @return a grid.
+     */
+    private int[][] mainDiagonalFlipHelper(int ring) {
+        int[][] resultGrid;
+        if (ring != 1) {
+            resultGrid = flipInternalRing(this.squarelotron);
+        } else {
+            resultGrid = flipExternalRing(this.squarelotron);
+        }
+        return resultGrid;
+    }
+
+    /**
+     * rotateRight.
+     *
+     * @param numberOfTurns number of turns.
+     */
+    void rotateRight(int numberOfTurns) {
+        final int maxNumberOfTurns = 4;
+        int totalTurns = numberOfTurns % maxNumberOfTurns;
+        if (totalTurns >= 0) {
+            rotateEast(numberOfTurns);
+            return;
+        }
+        rotateWest(numberOfTurns);
+    }
+
+    /**
+     * rotateEast.
+     *
+     * @param numberOfTurns number of turns.
+     */
+    private void rotateEast(int numberOfTurns) {
+        while (numberOfTurns > 0) {
+            for (int i = 1; i <= numberOfRings; i++) {
+                squarelotron = upsideDownFlipHelper(i);
+            }
+            for (int i = 1; i <= numberOfRings; i++) {
+                squarelotron = mainDiagonalFlipHelper(i);
+            }
+            numberOfTurns--;
+        }
+    }
+
+    /**
+     * rotateWest.
+     *
+     * @param numberOfTurns number of turns.
+     */
+    private void rotateWest(int numberOfTurns) {
+        while (numberOfTurns < 0) {
+            for (int i = 1; i <= numberOfRings; i++) {
+                squarelotron = mainDiagonalFlipHelper(i);
+            }
+            for (int i = 1; i <= numberOfRings; i++) {
+                squarelotron = upsideDownFlipHelper(i);
+            }
+            numberOfTurns++;
+        }
+    }
+
+    /**
+     * Prints the Squarelotron.
+     */
+    void printSquarelotron() {
+        for (int i = 0; i < squarelotron.length; i++) {
+            for (int j = 0; j < squarelotron.length; j++) {
+                System.out.print(" " + squarelotron[i][j] + " ");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)